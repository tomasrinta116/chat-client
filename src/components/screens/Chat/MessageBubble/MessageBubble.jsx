import React from "react";
import { withStyles } from "@material-ui/core/styles";
import styles from './MessageBubble.styles';
import PropTypes from 'prop-types';

const MessageBubble = props => {
  const {
    classes: { bubbleYou, infoMessage, bubbleOther, bubbleWrapper },
    sender,
    message
  } = props;
  let bubbleClass;
  let displayMessage;
  if (sender === "You") {
    bubbleClass = bubbleYou;
    displayMessage = `${sender}: ${message}`;
  } else if (sender === "info") {
    bubbleClass = infoMessage;
    displayMessage = `${message}`;
  } else {
    bubbleClass = bubbleOther;
    displayMessage = `${sender}: ${message}`;
  }
  return (
    <div className={bubbleWrapper}>
      <div id={'displayMessage'} className={bubbleClass}>{displayMessage}</div>
    </div>
  );
};

MessageBubble.propTypes = {
  sender: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
}

export default withStyles(styles)(MessageBubble);
