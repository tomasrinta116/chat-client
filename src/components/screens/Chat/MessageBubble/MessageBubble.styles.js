const styles = {
    bubbleWrapper: {
        width: "100%",
        display: "inline-flex", // sizes bubble to the text
        overflowWrap: "break-word",
        whiteSpace: "pre-wrap" // preserves whitespace
      },
      bubbleYou: {
        background: "#e3f2fd",
        color: "#1c54b2",
        borderRadius: 30,
        lineHeight: 1.3,
        marginBottom: "0.25rem",
        marginRight: "2rem",
        marginLeft: "auto", // pushes bubble to the right
        maxWidth: "60%",
        padding: 10
      },
      bubbleOther: {
        background: "#dbdbdb",
        color: "dark gray",
        borderRadius: 30,
        lineHeight: 1.3,
        marginBottom: "0.25rem",
        marginLeft: "2rem", // keeps bubble to the left
        maxWidth: 400,
        padding: 10
      },
      infoMessage: {
        margin: "auto",
        marginBottom: "0.5rem",
        marginTop: "0.5rem"
      },

}

export default styles; 
