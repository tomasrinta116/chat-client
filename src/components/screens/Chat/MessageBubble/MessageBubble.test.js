import React from "react";
import { shallow } from "enzyme";
import MessageBubble from "./MessageBubble";

describe("<MessagesDisplay />", () => {
  test("messages from 'sender: You' render correctly", () => {
    const props = {
      sender: "You",
      message: "test-you"
    };
    const wrapper = shallow(<MessageBubble {...props} />).dive();
    expect(wrapper).toMatchSnapshot();
  });

  test("messages from 'sender: someoneElse' render correctly", () => {
    const props = {
      sender: "someoneElse",
      message: "other-test"
    };
    const wrapper = shallow(<MessageBubble {...props} />).dive();
    expect(wrapper).toMatchSnapshot();
  });

  test("messages from 'info' render correctly", () => {
    const props = {
      sender: "info",
      message: "someone has left"
    };
    const wrapper = shallow(<MessageBubble {...props} />).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
