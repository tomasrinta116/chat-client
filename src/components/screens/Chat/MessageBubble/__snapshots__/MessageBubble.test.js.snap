// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessagesDisplay /> messages from 'info' render correctly 1`] = `
<div
  className="MessageBubble-bubbleWrapper-1"
>
  <div
    className="MessageBubble-infoMessage-4"
    id="displayMessage"
  >
    someone has left
  </div>
</div>
`;

exports[`<MessagesDisplay /> messages from 'sender: You' render correctly 1`] = `
<div
  className="MessageBubble-bubbleWrapper-1"
>
  <div
    className="MessageBubble-bubbleYou-2"
    id="displayMessage"
  >
    You: test-you
  </div>
</div>
`;

exports[`<MessagesDisplay /> messages from 'sender: someoneElse' render correctly 1`] = `
<div
  className="MessageBubble-bubbleWrapper-1"
>
  <div
    className="MessageBubble-bubbleOther-3"
    id="displayMessage"
  >
    someoneElse: other-test
  </div>
</div>
`;
